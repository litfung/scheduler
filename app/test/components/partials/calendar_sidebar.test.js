import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';

// import events from '../../__data__/events';
import groups from '../../__data__/groups';

import CalendarSidebarComp from '../../../src/client/components/partials/calendar/calendar_sidebar';
import configureStore from '../../../src/helpers/configure_store';

let wrapper, asyncFetchGroups, asyncPostGroup, asyncEditGroup, asyncDeleteGroup;

beforeEach(() => {
  asyncFetchGroups = jest.fn();
  asyncPostGroup = jest.fn();
  asyncEditGroup = jest.fn();
  asyncDeleteGroup = jest.fn();

  wrapper = shallow(
    <Provider store={configureStore({ get: jest.fn() })}>
      <CalendarSidebarComp
        visible={true}
        groups={groups}
        auth={{}}
        asyncFetchGroups={asyncFetchGroups}
        asyncPostGroup={asyncPostGroup}
        asyncEditGroup={asyncEditGroup}
        asyncDeleteGroup={asyncDeleteGroup}
      />
    </Provider>
  );
});

test('Snapshot for CalendarSidebarComp Component', () => {
  expect(wrapper).toMatchSnapshot();
});
