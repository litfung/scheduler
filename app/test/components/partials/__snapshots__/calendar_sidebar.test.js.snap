// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for CalendarSidebarComp Component 1`] = `
<Connect(CalendarSidebarComp)
  asyncDeleteGroup={[MockFunction]}
  asyncEditGroup={[MockFunction]}
  asyncFetchGroups={[MockFunction]}
  asyncPostGroup={[MockFunction]}
  auth={Object {}}
  groups={
    Array [
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c373f59d64892352c703256",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": true,
        "_rank": 0,
        "hex_color": "#4286f4",
        "title": "Tasks",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c3742106f0de60df8c62f82",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": false,
        "_rank": 3,
        "hex_color": "#f7ca18",
        "title": "Todos",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c38b60e5a84ae04e85e9d22",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": false,
        "_rank": 5,
        "hex_color": "#2ecc71",
        "title": "x",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c3fb10855de832ac8e374b2",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": false,
        "_rank": 10,
        "hex_color": "#2ecc71",
        "title": "New",
      },
    ]
  }
  visible={true}
/>
`;
